tests/integration_tests/conftest.py

```diff
@@ -216,6 +227,16 @@ def _client(request, fixture_utils, session_cloud: IntegrationCloud):
     if lxd_use_exec is not None:
         if not isinstance(session_cloud, _LxdIntegrationCloud):
             pytest.skip("lxd_use_exec requires LXD")
+        if isinstance(session_cloud, LxdVmCloud):
+            image_spec = ImageSpecification.from_os_image()
+            if image_spec.release == image_spec.image_id == "xenial":
+                # Why fail instead of skip?  We expect that skipped tests will
+                # be run in a different one of our usual battery of test runs
+                # (e.g. LXD-only tests are skipped on EC2 but will run in our
+                # normal LXD test runs).  This is not true of this test: it
+                # can't run in our usual xenial LXD VM test run, and it may not
+                # run anywhere else.  A failure flags up this discrepancy.
+                pytest.fail(XENIAL_LXD_VM_EXEC_MSG)
```

Comment:
I could go both ways on this. You make a good argument, but on the other hand, is this that much different from what we're doing for other tests? We skip on unsupported configurations. The only way we know a test is going to get run is if we make sure that it's included in the configuration matrix of tests we're going to run. If we know we have tests that require exec, we should probably include an exec-able image to our default test runs.

But I'm ok going with this. If it causes frustration, it wouldn't be hard to change later.